src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/ModelMetadata.cs

```diff
@@ -48,6 +48,12 @@ public class ModelMetadata
         }
 
         /// <summary>
+        /// Gets a collection of additional information about the model.
+        /// </summary>
+        public virtual IDictionary<string, object> AdditionalValues { get; }
```

Comment:
/cc @yishaigalatzer we should think about `<object, object>` instead, that's what we're doing in other places where we have arbitrary user-defined properties (apidescription, actiondescriptor)